app/main.py

```diff
@@ -1,2 +1,68 @@
+from typing import Any
+
+
+class Node:
+    def __init__(self, key: Any, value: Any) -> None:
+        self.key = key
+        self.value = value
+        self.next = None
+
+
 class Dictionary:
-    pass
+    def __init__(
+            self, initial_capacity: float = 16,
+            load_factor: float = 0.75
+    ) -> None:
+        self.capacity = initial_capacity
+        self.load_factor = load_factor
+        self.size = 0
+        self.table = [None] * self.capacity
+
+    def _hash_function(self, key: Any) -> float:
+        return hash(key) % self.capacity
+
+    def _resize(self, new_capacity: Any) -> None:
+        old_table = self.table
+        self.table = [None] * new_capacity
+        self.capacity = new_capacity
+        self.size = 0
+
+        for node in old_table:
+            while node:
+                self[node.key] = node.value
+                node = node.next
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        hash_value = self._hash_function(key)
+        new_node = Node(key, value)
+
+        if self.table[hash_value] is None:
+            self.table[hash_value] = new_node
+        else:
+            current_node = self.table[hash_value]
+            while current_node:
+                if current_node.key == key:
```

Comment:
Also check hash

Reply:
You must check for the key
current_node.key == key and current_node.hash == key_hash